paddle/fluid/framework/details/build_strategy.cc

```diff
@@ -145,6 +161,20 @@ std::unique_ptr<ir::Graph> BuildStrategy::Apply(
       pass->Erase("nccl_ctxs");
       pass->SetNotOwned<platform::NCCLContextMap>("nccl_ctxs", nctx);
 #endif
+    } else if (pass->Type() == "analysis_var_pass") {
+      const std::vector<OpDesc *> *all_op_descs =
+          new std::vector<OpDesc *>(main_program.Block(0).AllOps());
+      GraphNodePool *graph_pool = new GraphNodePool;
+      graph->Set<const std::vector<OpDesc *>>(kAllOpDescs,
+                                              all_op_descs);  // take ownership
+      graph->Set<const GraphNodePool>(kGraphNodePool,
+                                      graph_pool);  // take ownership
+
+      pass->Erase(kAllOpDescs);
+      pass->SetNotOwned<const std::vector<OpDesc *>>(kAllOpDescs, all_op_descs);
+      pass->Erase(kGraphNodePool);
+      pass->SetNotOwned<GraphNodePool>(kGraphNodePool, graph_pool);
```

Comment:
Why both graph and pass are set kGraphNodePool? It seems graph is enough?

Reply:
polish code.